azure-self-hosted-runners/post-deployment-script.ps1

```diff
@@ -28,10 +28,51 @@ Write-Output "Starting post-deployment script."
 # =================================
 # TOOL VERSIONS AND OTHER VARIABLES
 # =================================
+#
+# This header is used for both Git for Windows and GitHub Actions Runner
+[hashtable]$GithubHeaders = @{
+    "Accept"               = "application/vnd.github.v3+json"
+    "X-GitHub-Api-Version" = "2022-11-28"
+}
 
-$GitForWindowsVersion = "2.39.0"
-$GitForWindowsTag = "2.39.0.windows.1"
-$GitForWindowsHash = "2eaba567e17784654be77ba997329742d87845c6f15e33c9620f9a331c69a976"
+# =================================
+# Get download and hash information for the latest release of Git for Windows
+# =================================
+#
+# This will return the latest release of Git for Windows download link, hash and the name of the outfile
+# Everything will be saved in the object $GitHubGit
+#
+# url for Github API to get the latest release
+[string]$GitHubUrl = "https://api.github.com/repos/git-for-windows/git/releases/latest"
+#
+# Name of the exe file that should be verified and downloaded
+[string]$GithubExeName = "Git-.*-64-bit.exe"
+
+try {
+    [System.Object]$GithubRestData = Invoke-RestMethod -Uri $GitHubUrl -Method Get -Headers $GithubHeaders -TimeoutSec 10 | Select-Object -Property assets, body
+    [System.Object]$GitHubAsset = $GithubRestData.assets | Where-Object { $_.name -match $GithubExeName }
+    if ($GithubRestData.body -match "\b${[Regex]::Escape($GitHubAsset.name)}.*?\|.*?([a-zA-Z0-9]{64})" -eq $True) {
+        [System.Object]$GitHubGit = [PSCustomObject]@{
+            DownloadUrl = [string]$GitHubAsset.browser_download_url
+            Hash        = [string]$Matches[1].ToUpper()
```

Comment:
I learned something new today: I had not known that `$Matches` is set automatically by `-match`. Thank you!

Reply:
:)